packages/api/amplify_api/test/util.dart

```diff
@@ -60,3 +67,111 @@ void validateSignedRequest(http.BaseRequest request) {
     contains('aws-sigv4'),
   );
 }
+
+const testApiKeyConfig = AWSApiConfig(
+  endpointType: EndpointType.graphQL,
+  endpoint: 'https://abc123.appsync-api.us-east-1.amazonaws.com/graphql',
+  region: 'us-east-1',
+  authorizationType: APIAuthorizationType.apiKey,
+  apiKey: 'abc-123',
+);
+
+const expectedApiKeyWebSocketConnectionUrl =
+    'wss://abc123.appsync-realtime-api.us-east-1.amazonaws.com/graphql?header=eyJDb250ZW50LVR5cGUiOiJhcHBsaWNhdGlvbi9qc29uOyBjaGFyc2V0PVVURi04IiwiWC1BcGktS2V5IjoiYWJjLTEyMyIsIkFjY2VwdCI6ImFwcGxpY2F0aW9uL2pzb24sIHRleHQvamF2YXNjcmlwdCIsIkNvbnRlbnQtRW5jb2RpbmciOiJhbXotMS4wIiwiSG9zdCI6ImFiYzEyMy5hcHBzeW5jLWFwaS51cy1lYXN0LTEuYW1hem9uYXdzLmNvbSJ9&payload=e30%3D';
+
+AmplifyAuthProviderRepository getTestAuthProviderRepo() {
+  final testAuthProviderRepo = AmplifyAuthProviderRepository();
+  testAuthProviderRepo.registerAuthProvider(
+    APIAuthorizationType.apiKey.authProviderToken,
+    AppSyncApiKeyAuthProvider(),
+  );
+
+  return testAuthProviderRepo;
+}
+
+const mockSubscriptionData = {
+  'onCreatePost': {
+    'id': '49d54440-cb80-4f20-964b-91c142761e82',
+    'title':
+        'Integration Test post - subscription create aa779f0d-0c92-4677-af32-e43f71b3eb55',
+    'rating': 0,
+    'created': null,
+    'createdAt': '2022-08-15T18:22:15.410Z',
+    'updatedAt': '2022-08-15T18:22:15.410Z',
+    'blog': {
+      'id': '164bd1f1-544c-40cb-a656-a7563b046e71',
+      'name': 'Integration Test Blog with a post - create',
+      'createdAt': '2022-08-15T18:22:15.164Z',
+      'file': null,
+      'files': null,
+      'updatedAt': '2022-08-15T18:22:15.164Z'
+    }
+  }
+};
+
+/// Extension of [WebSocketConnection] that stores messages internally instead
+/// of sending them.
+class MockWebSocketConnection extends WebSocketConnection {
```

Comment:
At some point we should be mocking/injecting the `WebSocketChannel` instead and not overriding methods since it becomes unclear what's being tested. Like here, I guess we're testing everything but `connect`/`send` - but those seem like good things to test too.

Reply:
makes sense, I put a task in backlog to improve this